### PR TITLE
Spring boot

### PR DESCRIPTION
**Pull request by NatashaBel**
Screenshot 2024-10-31 at 18 30 27

### CODE REVIEW

File: src/main/java/org/example/repository/TicketRepository.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import java.util.List;
+import java.util.UUID;
+
+@Repository
+public interface TicketRepository extends CrudRepository<Ticket, UUID> {
+    List<Ticket> findByUserId(UUID userId);
+
+    @Modifying
+    @Transactional
```
Comment (AtticSiol) [inline src/main/java/org/example/repository/TicketRepository.java]: move this `@Transactional` to service layer: in spring mvc apps services handle business logic, often involving multiple repositories. Placing transactions here ensures atomicity across multiple operations.

File: src/main/java/org/example/repository/UserRepository.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Repository;
+
+import java.util.UUID;
+
+@Repository
+public interface UserRepository extends CrudRepository<User, UUID> {
+    @Modifying
+    @Transactional
```
Comment (AtticSiol) [inline src/main/java/org/example/repository/UserRepository.java]: move this @Transactional to service layer

File: src/main/java/org/example/controller/TicketController.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+@RequestMapping("/api/tickets")
+public class TicketController {
+
+    @Autowired
+    private String thisIsMyFirstConditionalBean;
+
+    @Autowired
+    private TicketService ticketService;
+
+    @PostMapping("/")
```
Comment (AtticSiol) [inline src/main/java/org/example/controller/TicketController.java]: better to use just `@PostMapping` without arguments if the url is equal to url in `@RequestMapping`

File: src/main/java/org/example/controller/TicketController.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        return ticketService.saveTicket(ticket);
+    }
+
+    @GetMapping("/{id}")
+    public Ticket getTicketById(@PathVariable UUID id) {
+        return ticketService.getTicket(id)
+                .orElseThrow(() -> new IllegalArgumentException("Ticket not found with id: " + id));
+    }
+
+    @GetMapping("/user/{userId}")
```
Comment (AtticSiol) [inline src/main/java/org/example/controller/TicketController.java]: `users` - always in plural form

File: src/main/java/org/example/controller/TicketController.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    public List<Ticket> getTicketsByUserId(@PathVariable UUID userId) {
+        return ticketService.getTicketByUserId(userId);
+    }
+
+    @PutMapping("/{id}/type")
```
Comment (AtticSiol) [inline src/main/java/org/example/controller/TicketController.java]: perfectly will be `@PutMapping("/{id}/update-type")`

File: src/main/java/org/example/controller/TicketController.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+    @PostMapping("/")
+    public Ticket createTicket(@RequestBody Ticket ticket) {
+        return ticketService.saveTicket(ticket);
+    }
+
+    @GetMapping("/{id}")
+    public Ticket getTicketById(@PathVariable UUID id) {
+        return ticketService.getTicket(id)
+                .orElseThrow(() -> new IllegalArgumentException("Ticket not found with id: " + id));
```
Comment (AtticSiol) [inline src/main/java/org/example/controller/TicketController.java]: move logic about error handling to service layer. on controller layer exceptions will be handled with the help of `@ExceptionHandler` and `@ControllerAdvice`

Comment (NatashaBel) [reply src/main/java/org/example/controller/TicketController.java]: Moved logic about error handling to service layer.

File: src/main/java/org/example/controller/TicketController.java (hunk 5 above)
Comment (NatashaBel) [reply src/main/java/org/example/controller/TicketController.java]: Changed to @PutMapping("/{id}/update-type")

File: src/main/java/org/example/controller/TicketController.java (hunk 4 above)
Comment (NatashaBel) [reply src/main/java/org/example/controller/TicketController.java]: Changed  from  @GetMapping("/user/{userId}") to  @GetMapping("/users/{userId}")

File: src/main/java/org/example/controller/TicketController.java (hunk 3 above)
Comment (NatashaBel) [reply src/main/java/org/example/controller/TicketController.java]: Deleted ("/")

File: src/main/java/org/example/repository/UserRepository.java (hunk 2 above)
Comment (NatashaBel) [reply src/main/java/org/example/repository/UserRepository.java]: Moved @Transactional to service layer

File: src/main/java/org/example/repository/TicketRepository.java (hunk 1 above)
Comment (NatashaBel) [reply src/main/java/org/example/repository/TicketRepository.java]: Moved @Transactional to service layer

**Review by AtticSiol — APPROVED** [review]

**Pull request merged**